frontend/components/Documents.js

```diff
@@ -56,6 +57,25 @@ const Documents = () => {
 	    }));
     };
 
+    const handleAttributeInputChange = (event, index) => {
+        const {name, value} = event.target;
+        setNewAttributes(prevAttributes => {
+            return prevAttributes.map((attribute, i) => {
+                return i === index ? {...attribute, [name]:value} : attribute;
+            });
+        });
+    };
+
+    const handleAddAttribute = () => {
+        setNewAttributes([...newAttributes, {"name": "", "value": ""}]);
+    };
+
+    const handleRemoveAttribute = (index) => {
+        const attributesList = [...newAttributes];
+        attributesList.splice(index, 1);
+        setNewAttributes(attributesList);
```

Comment:
I'm torn on this. This definitely works, but I paused at it for a second because I thought it fell into the same trap as above: i.e., we're spreading objects into a new array, so they're being copied by reference (not deep copied), therefore we run the risk of directly modifying state. That said, since we're splicing a newly created array, we're not actually modifying the underlying objects in any way, so this should definitely just work.

The alternative would look something like:

```
setNewAttributes(
  (previousAttributes) => previousAttributes.filter((attribute, attribute_index) => attribute_index !== index)
);
```